### DISCUSSION TITLE
Devika for existing code base

### DISCUSSION THREAD

**Discussion by lesego94** [stitionai/devika#315] [Q&A]
I am very new to Devika, could I use it on an existing local codebase to modify existing code and run tests?

**Comment by Whytehorse2022** [comment]
Give it a try. You probably need to specify the path to the project.

**Comment by samikrc** [comment]
I am interested in this too. Do you mean specifying the path to the project in the instructions? Or is there a parameter at the start-up time? The second one would be helpful - if the system can limit the context to a single project, and can possibly produce better quality responses.

**Reply by Whytehorse2022** [reply]
So you have two prompts with LLMs. The system and user. You can change either one if you have access. That's where you would specify a path. "Hey Devika, here is my code base: /user/home/me/codebase. All following questions will be about that." If you put it in the system prompt it will always know that, otherwise you have to keep specifying. There's also a problem called "needle in a haystack" where the LLMs forget the middle of a very long text. Google claims to have solved it. I don't know if Devika has it solved.
